### ISSUE TITLE
Freezing on Kobo Nia when no wifi connection

### ISSUE THREAD

**Issue by themaz** [koreader/koreader#10558]
* KOReader version: v2023.05.1-9-g18cd6b79_2023-06-05
* Device: Kobo Nia

#### Issue
Koreader consistently freezes on the Kobo Nia when not connected to wifi. The freeze seems to happen about 2-5 minutes after restart. I know Kobo Nia is not one of the fully supported devices, but hoping this crash log can help debug the freezing issue on the device.

#### Steps to reproduce
- Open Koreader
- Open a book
- Ensure you're not connected to wifi
- Read until Koreader freezes

##### `crash.log` (if applicable)
[crash.log](https://github.com/koreader/koreader/files/11709250/crash.log)

**Comment by NiLuJe** [comment]
The Nia runs on a similar board as the Libra 2 & Clara 2E, so I suspect you're seeign the exact same thing as #8414 & #9806.

TL;DR: board/kernel issue, don't have the hardware to investigate. Disabling Notifications seems like the best workaround so far.

**Comment by NiLuJe** [comment]
Speaking of not having the hardware, nobody got back to me about testing https://github.com/koreader/koreader/issues/9806#issuecomment-1566292305, FWIW.

**Comment by maziar-mastermind** [comment]
Thanks @NiLuJe - FYI I have disabled all notifications, but that didn't seem to help. Regarding testing #9806 - can you elaborate a bit? Should I remove the `if-statement` in that file?

**Comment by themaz** [comment]
Thanks @NiLuJe - FYI I have disabled all notifications, but that didn't seem to help. Regarding testing https://github.com/koreader/koreader/issues/9806 - can you elaborate a bit? Should I remove the if-statement in that file?

Reposted this with my correct GH account.

**Comment by NiLuJe** [comment]
> Should I remove the if-statement in that file?

And the matching `end`, yeah.

**Comment by themaz** [comment]
Thanks - did this last night and the Nia still crashed after about 2 minutes when I wasn't connected to a wifi network.

**Comment by NiLuJe** [comment]
Let's hope https://github.com/koreader/koreader/pull/10771 did the trick ;).

**Comment by leanne-purple** [comment]
> Let's hope https://github.com/koreader/koreader/pull/10771 did the trick ;).

How do you use https://github.com/koreader/koreader/pull/10771 ?
I'm on kobo nia and getting theses crashes

**Comment by Frenzie** [comment]
It's included in any release from the past two years. By these crashes you mean it freezes without doing anything? Could you also provide a crash.log after it happens?

**Comment by leanne-purple** [comment]
The crashes are very random, I can be in the middle of turning a page, or just leaving it alone and it completely freezes, forcing me to restart it

**Comment by leanne-purple** [comment]
> It's included in any release from the past two years. By these crashes you mean it freezes without doing anything? Could you also provide a crash.log after it happens?

[crash.log](https://github.com/user-attachments/files/26384359/crash.log)

**Comment by Frenzie** [comment]
Right, so in the sense that there's nothing to see that does indeed look the same. Pinging @NiLuJe on the off chance some time presents itself.

**Comment by NiLuJe** [comment]
I vaguely remember that there is *one* of the affected devices that *does not* support the workaround, and I'm wondering if that might not be the Nia...

EDIT: 5 minutes of trying to piece it together from my notes later, I noticed that I'd just written it down in the [linked issue](https://github.com/koreader/koreader/pull/10771#issuecomment-1885428277), lol.

TL;DR: Yah, the Nia sucks, unfortunately ;/.

**Comment by leanne-purple** [comment]
> I vaguely remember that there is *one* of the affected devices that *does not* support the workaround, and I'm wondering if that might not be the Nia...
> 
> EDIT: 5 minutes of trying to piece it together from my notes later, I noticed that I'd just written it down in the [linked issue](https://github.com/koreader/koreader/pull/10771#issuecomment-1885428277), lol.
> 
> TL;DR: Yah, the Nia sucks, unfortunately ;/.

Soon there's nothing we can do ? 😞

**Comment by NiLuJe** [comment]
Not that I know of, no.

(And the device itself is relatively unpopular, given its weird spot in the lineup: there's not enough incentive to go for it instead of the better Clara; so that doesn't help. Not many devices in the wild (especially in our nerdy niche), and even less people to work with it ;)).
